ios/browser/api/brave_shields/adblock_engine.mm

```diff
@@ -77,67 +72,50 @@ - (NSString*)cspDirectivesForURL:(NSString*)url
-      base::SysNSStringToUTF8(resourceType)));
+      base::SysNSStringToUTF8(tabHost), base::SysNSStringToUTF8(resourceType),
+      isThirdParty));
 }
 
 - (bool)deserialize:(NSData*)data {
-  return adblock_engine->deserialize(static_cast<const char*>(data.bytes),
-                                     data.length);
+  auto result = adblock_engine->deserialize(
+      static_cast<const char*>(data.bytes), data.length);
+  return result.result_kind == adblock::ResultKind::Success;
 }
 
 - (void)addTag:(NSString*)tag {
-  adblock_engine->addTag(base::SysNSStringToUTF8(tag));
+  adblock_engine->enable_tag(base::SysNSStringToUTF8(tag));
 }
 
 - (void)removeTag:(NSString*)tag {
-  adblock_engine->removeTag(base::SysNSStringToUTF8(tag));
+  adblock_engine->disable_tag(base::SysNSStringToUTF8(tag));
 }
 
 - (bool)tagExists:(NSString*)tag {
-  return adblock_engine->tagExists(base::SysNSStringToUTF8(tag));
+  return adblock_engine->tag_exists(base::SysNSStringToUTF8(tag)).value;
 }
 
 - (void)addResourceWithKey:(NSString*)key
                contentType:(NSString*)contentType
                       data:(NSString*)data {
-  adblock_engine->addResource(base::SysNSStringToUTF8(key),
-                              base::SysNSStringToUTF8(contentType),
-                              base::SysNSStringToUTF8(data));
+  adblock_engine->add_resource(base::SysNSStringToUTF8(key),
+                               base::SysNSStringToUTF8(contentType),
+                               base::SysNSStringToUTF8(data));
 }
 
 - (void)useResources:(NSString*)resources {
-  adblock_engine->useResources(base::SysNSStringToUTF8(resources));
+  adblock_engine->use_resources(base::SysNSStringToUTF8(resources));
 }
 
 - (NSString*)cosmeticResourcesForURL:(NSString*)url {
   return base::SysUTF8ToNSString(
-      adblock_engine->urlCosmeticResources(base::SysNSStringToUTF8(url)));
+      adblock_engine->url_cosmetic_resources(base::SysNSStringToUTF8(url))
+          .value);
 }
 
-- (NSString*)
-    stylesheetForCosmeticRulesIncludingClasses:(NSArray<NSString*>*)classes
-                                           ids:(NSArray<NSString*>*)ids
-                                    exceptions:(NSArray<NSString*>*)exceptions {
-  return base::SysUTF8ToNSString(adblock_engine->hiddenClassIdSelectors(
-      brave::ns_to_vector<std::string>(classes),
-      brave::ns_to_vector<std::string>(ids),
-      brave::ns_to_vector<std::string>(exceptions)));
-}
-
-+ (bool)setDomainResolver:(DomainResolverCallback)domainResolver {
-  return adblock::SetDomainResolver(domainResolver);
-}
-
-+ (DomainResolverCallback)defaultDomainResolver {
-  return brave_shields::AdBlockServiceDomainResolver;
-}
-
-+ (NSString*)contentBlockerRulesFromFilterSet:(NSString*)filterSet {
-  return base::SysUTF8ToNSString(adblock::ConvertRulesToContentBlockingRules(
-      base::SysNSStringToUTF8(filterSet)));
++ (bool)setDomainResolver {
+  return adblock::set_domain_resolver(domainResolver);
 }
```

Comment:
This won't compile, why was the argument removed here?

Reply:
Right, this is where i might need some help from you guys. The argument is removed because we set the engine domain resolver to a wrapper func that calls the cpp domain resolver (the cpp func is directly linked).

Reply:
https://github.com/brave/brave-core/pull/17368/files#diff-43376c43e4f46217cf364042ddf0696e5f65941d0e9b47b783d8137617edf9b4R46-R58

Reply:
https://github.com/brave/brave-core/blob/df4cc08487fa0f56267b7ffc30ec7c6c6d8f7671/components/brave_shields/adblock/rs/src/engine.rs#L46-L58

Reply:
sidenote, passing functions to Rust land is an open issue in cxx: dtolnay/cxx#52

Reply:
> the cpp func is directly linked

Direct linking is probably the best option, if possible. Hadn't occurred to me when I wrote it that it was an option, but I 100% support changing to that if we can.

Reply:
Ah so if I understand correctly we no longer need to set this to `brave_shields::AdBlockServiceDomainResolver` ourselves and we can just call  `adblock::set_domain_resolver` with an empty arg at least once somewhere? Can this be called multiple times?

Reply:
Correct